airflow/hooks/hive_hooks.py

```diff
@@ -614,62 +614,96 @@ def get_partitions(
             return [dict(zip(pnames, p.values)) for p in parts]
 
     @staticmethod
-    def _get_max_partition_from_part_names(part_names, key_name):
+    def _get_max_partition_from_part_names(part_names, partition_key, filter_map):
         """
-        Helper method to get max partition from part names. Works only
-        when partition format follows '{key}={value}' and key_name is name of
-        the only partition key.
-        :param part_names: list of partition names
+        Helper method to get max partition of partitions with partition_key
+        from part names. key:value pair in filter_map will be used to
+        filter out partitions.
+
+        :param part_names: list of partition names.
         :type part_names: list
-        :param key_name: partition key name
-        :type key_name: str
+        :param partition_key: partition key names.
+        :type partition_key: string
+        :param filter_map: partition_key:partition_value map used for partition filtering.
+        :type filter_map: map
         :return: Max partition or None if part_names is empty.
         """
         if not part_names:
             return None
 
-        prefix = key_name + '='
-        prefix_len = len(key_name) + 1
-        max_val = None
+        part_dicts = []
         for part_name in part_names:
-            if part_name.startswith(prefix):
-                if max_val is None:
-                    max_val = part_name[prefix_len:]
-                else:
-                    max_val = max(max_val, part_name[prefix_len:])
-            else:
-                raise AirflowException(
-                    "Partition name mal-formatted: {}".format(part_name))
-        return max_val
+            part_dict = {}
+            for single_part_name in part_name.split('/'):
```

Comment:
There's a slight edge case here - if there's a % in the name, it needs to be decoded by a call to the Hive metastore.